commit 92bc2b5bc01fe53a18f2ec03098a757ebf78ac13
Author: PJBrs <pjbrs@floorenpj.xs4all.nl>
Date:   Fri Feb 21 09:00:21 2020 +0100

    ARM: dts: sunxi: h2-plus-bananapi-m2-zero: Bluetooth and wifi rfkill
    
    Add gpios for bluetooth rfkill and something with wifi. Now
    I'm definitely out of my depth.

diff --git a/arch/arm/boot/dts/sun8i-h2-plus-bananapi-m2-zero.dts b/arch/arm/boot/dts/sun8i-h2-plus-bananapi-m2-zero.dts
index 0b702168981c..146c1a9c4335 100644
--- a/arch/arm/boot/dts/sun8i-h2-plus-bananapi-m2-zero.dts
+++ b/arch/arm/boot/dts/sun8i-h2-plus-bananapi-m2-zero.dts
@@ -47,6 +47,17 @@
 		};
 	};
 
+	rfkill_bt: rfkill_bt {
+		compatible = "rfkill-gpio";
+		pinctrl-names = "default";
+		pinctrl-0 = <&bt_pwr_pin>;
+		reset-gpios = <&pio 6 12 GPIO_ACTIVE_HIGH>; /* PG12 */
+		clocks = <&osc32k>;
+		clock-frequency = <32768>;
+		rfkill-name = "sunxi-bt";
+		rfkill-type = "bluetooth";
+	};
+
 	gpio_keys {
 		compatible = "gpio-keys";
 
@@ -81,6 +92,21 @@
 	};
 };
 
+&pio {
+	bt_pwr_pin: bt_pwr_pin {
+		pins = "PG12";
+		function = "gpio_out";
+	};
+};
+
+
+&r_pio {
+	wifi_en_pin: wifi_en_pin {
+		pins = "PL7";
+		function = "gpio_out";
+	};
+};
+
 &cpu0 {
         cpu-supply = <&reg_vdd_cpux>;

